popup/Calendar/CreateCalendar.md

```diff
@@ -0,0 +1,284 @@
+
+/*The date format used when the calendar is not displayed. See https://developers.arcgis.com/arcade/function-reference/text_functions/#text  */
+var date_format = "MMMM D, Y"
+
+/*Specify the number of months to display. Only applicable if show_calendar is true*/
+var num_months = 2
+
+/*Indicate color of header on calendar. Only applicable if show_calendar is true*/
+var header_color="#1a6324"
+
+/*Indicate color of the text on the calendar's header. Only applicable if show_calendar is true*/
+var header_text_color = "#FFFFFF"
+
+/*Indicate color of "today" outline on calendar. Only applicable if show_calendar is true*/
+var today_color = "#8c8c8c"
+
+/*Indicate default max width of each calendar*/
+var max_width = "49%"
+
+/*Indicate the minimum width of each calendar */
+var min_width = "250px"
+
+
+
+/*DO NOT CHANGE ANYTHING BELOW THIS LINE
+-------------------------------------------------------------------------*/
+
+function CreateCalendar(events,showCalendar,displayText,dateFormat,numberMonths,headerColor,headerTextColor,todayColor,minWidth,maxWidth){
+ 	//Check inputs and set defaults
+	if(IsEmpty(events)){
+		return("Event List Required")
+	}
+
+	iif(IsEmpty(showCalendar),showCalendar=True,showCalendar)
+	iif(IsEmpty(numberMonths),numberMonths=2,numberMonths)
+	iif(IsEmpty(displayText),displayText="The next scheduled event is ",displayText)
+	iif(IsEmpty(headerColor),headerColor="#007AC2",headerColor)
+	iif(IsEmpty(headerTextColor),headerTextColor="#FFFFFF",headerTextColor)
+	iif(IsEmpty(todayColor),todayColor="#6A6A6A",todayColor)
+	iif(IsEmpty(minWidth),minWidth="250px",minWidth)
+	iif(IsEmpty(maxWidth),maxWidth="49%",maxWidth)
+	iif(IsEmpty(dateFormat),dateFormat = "MMMM D, Y",dateFormat)
+
+
+	var temp = events
+	var eventColors = {}
+	events = {}
+	for (var y in temp){
+		eventColors[temp[y]["Name"]]=temp[y]["Color"]
+		events[temp[y]["Name"]]=[]
+		for (var d in temp[y]["Events"]){
+			push(events[temp[y]["Name"]],date(temp[y]["Events"][d]))
+		}
+	}
+
+
+
+	var m_days={"January":31,"February":28,"March":31,"April":30,"May":31,"June":30, "July":31,"August":31,"September":30,"October":31,"November":30,"December":31}
+	var m_number={"January":0,"February":1,"March":2,"April":3,"May":4,"June":5, "July":6,"August":7,"September":8,"October":9,"November":10,"December":11}
+	var m_name ={"0":"January","1":"February","2":"March","3":"April","4":"May","5":"June","6":"July","7":"August","8":"September","9":"October","10":"November","11":"December"}
+	var EventDays = {};
+
+	if ((number(text(today(),"Y")) % 4 == 0)){
+			m_days["February"]=29
+	}
+
+
+	var key_list=[]
+	var final_text
+	
+	//Create Months
+	var months = dictionary(text(today(),"MMMM"),array(m_days[text(today(),"MMMM")]))
+	var cycle_months = array(0)
+	push(cycle_months,text(today(),"M"))
+
+	for (var i = 0; i < numberMonths; i++) {
+			if(year(today())<year(Dateadd(today(),i,"months")) && text(Dateadd(today(),i,"months"),"MMMM")== "February"){
```

Comment:
you are calling the function today a lot, maybe store as variable, not sure the performance hit, but why keep calling it